packages/advanced-logic/specs/payment-network-erc20-fee-proxy-contract-0.1.0.md

```diff
@@ -0,0 +1,328 @@
+| **parameters.feeAmount**      | `feeAmount` from parameters if given, undefined otherwise      |
+| **parameters.salt**           | Salt for the request                                           |
+
+---
+
+### Updates
+
+#### addPaymentAddress
+
+##### Parameters
+
+|                               | Type   | Description                                   | Requirement   |
+| ----------------------------- | ------ | --------------------------------------------- | ------------- |
+| **id**                        | String | Constant value: "pn-erc20-fee-proxy-contract" | **Mandatory** |
+| **action**                    | String | Constant value: "addPaymentAddress"           | **Mandatory** |
+| **parameters**                | Object |                                               |               |
+| **parameters.paymentAddress** | String | Ethereum address for the payment              | **Mandatory** |
+
+##### Conditions
+
+This action is valid, if:
+
+- The extension state with the id "pn-erc20-fee-proxy-contract" exists
+- The signer is the `payee`
+- The extension property `paymentAddress` is undefined
+
+##### Warnings
+
+None.
+
+##### Results
+
+An extension state is updated with the following properties:
+
+|  Property                 |  Value                                               |
+| ------------------------- | ---------------------------------------------------- |
+| **values.paymentAddress** | `paymentAddress` from parameters                     |
+| **events**                | Add an 'paymentAddress' event (see below) at its end |
+
+the 'addPaymentAddress' event:
+
+|  Property                     |  Value                              |
+| ----------------------------- | ----------------------------------- |
+| **name**                      | Constant value: "addPaymentAddress" |
+| **parameters**                |                                     |
+| **parameters.paymentAddress** | `paymentAddress` from parameters    |
+
+#### addRefundAddress
+
+##### Parameters
+
+|                              | Type   | Description                                   | Requirement   |
+| ---------------------------- | ------ | --------------------------------------------- | ------------- |
+| **id**                       | String | Constant value: "pn-erc20-fee-proxy-contract" | **Mandatory** |
+| **action**                   | String | Constant value: "addRefundAddress"            | **Mandatory** |
+| **parameters**               | Object |                                               |               |
+| **parameters.refundAddress** | String | Ethereum address for the refund               | **Mandatory** |
+
+##### Conditions
+
+This action is valid if:
+
+- The extension state with the id "pn-erc20-fee-proxy-contract" exists
+- The signer is the `payer`
+- The extension property `refundAddress` is undefined
+
+##### Warnings
+
+None.
```

Comment:
Shouldn't there be a warning if the payee tries to add a refund address?

Reply:
Same as previous, the conditions will make sure to filter this case out.